### ISSUE TITLE
[SUPPORT] Add AS OF syntax support

### ISSUE THREAD

**Issue by melin** [apache/iceberg#4365]
Use sql to query the specified version data
``` 
SELECT * FROM default.people10m VERSION AS OF 0; 
SELECT * FROM default.people10m TIMESTAMP AS OF '2019-01-29 00:37:58'; 
```

**Comment by singhpk234** [comment]
Hello All members,

Wanted to check if community is interested in taking this feature in for spark sql (looks like trino already has an open pr https://github.com/trinodb/trino/pull/10258 for it), if yes is some one actively working on this, if not I am happy to take this up.

**Comment by rdblue** [comment]
Fixed by #5156
